### PR TITLE
Adding first community repo labels

### PR DESCRIPTION
**Pull request by quaid**
Labels for issues and PRs related to community needs. These are being included in the default labels because I'm pretty sure that activity across the project will be part of the community aspects these labels represent.

As these are my first set of labels going into this automation, I won't be surprised if I did something incorrectly or a bit too much (like, did I have to add new labels in manually, or does the bot sort by alpha itself later? etc.)

Signed-off by:  kwade@redhat.com, quaid@iquaid.org

## Related Issues and Dependencies

…

## Does this require new deployment ?

- [ ] Deployment for Test and Stage `AICoE/aicoe-cd` and Prod `operate-first/argocd-apps`.

### CODE REVIEW

**Comment by quaid** [comment]
I seem to have done this prematurely, I now notice there are two humans who need to review this first.

/unassign @goern

File: prow/overlays/cnv-prod/labels_operate-first.yaml
Comment (goern) [inline prow/overlays/cnv-prod/labels_operate-first.yaml:366]: This might be equivalent to `kind/documentation` ?

**Comment by goern** [comment]
/ok-to-test

@quaid just one small question and the note, that Thoth Station is working with one approver ;)

**Comment by goern** [comment]
@quaid to prevent the pre-commit error, use `pre-commit install` in your local clone of the repo...

File: prow/overlays/cnv-prod/labels_operate-first.yaml
Comment (quaid) [reply prow/overlays/cnv-prod/labels_operate-first.yaml:366]: I think it depends on the full scope of `kind/documentation`, but I suspect there is some overlap and some not.

From a project perspective, we will always have a set of users and a growing body of documentation for them. The definition of 'users' includes:  open source projects doing QE; people on learning pathways (SRE, data scientist, etc.); people who have other workloads on the system for any reason (running demos, etc.) Thus the purpose of the `community/documentation` label is to be solely for documentation-related issues/PRs that are intended to be part of the user-facing documentation. (I am intentionally not limiting it to written content, as it may be video, audio, and other artifacts, and it may be related to tooling and automation to bring the learning pathways to life—anything being done in the name of "end-user documentation".) 

I suspect that `kind/documentation` covers documentation for the running cloud environment and other cases. If that's the case, then `community/documentation` is a subset of that set, but with a twist:

It's almost certain that some portion of the "documentation for keeping the cloud environment running" will also be auto-included in the learning pathways for "people learning how this thing is running" and especially for "people learning hands-on how this is running by submitting a PR on the live system they can watch get tested, accepted, and deployed." But I don't think all of the docs and tooling that might be `kind/documentation` is going to be part of end-user documentation, so there is value in having two distinct but intertwinable labels.

Comment (goern) [reply prow/overlays/cnv-prod/labels_operate-first.yaml:366]: Right, got it. What about combining labels? `kind/documentation` `area/community` is for what you intended, `kind/bug` `kind/documentation` could be for doc bugs, and `kind/feature` `area/community` could be used for a new feature that we want to implement for the community?

In the end, its all up to you... To me it just felt handy to reuse what we got, and as subjects get contextualized in different ways, multiple labels might be a good solution.

wdyt?

**Comment by quaid** [comment]
> @quaid to prevent the pre-commit error, use `pre-commit install` in your local clone of the repo...

I've been cheating and using the web UI :grin: 

Can you point me at the current contributor or user docs that cover the `pre-commit` requirement?

File: prow/overlays/cnv-prod/labels_operate-first.yaml
Comment (quaid) [reply prow/overlays/cnv-prod/labels_operate-first.yaml:366]: OK, I think I see where you are going. I definitely want to create metadata that fits the format of other labels and helps machine learning, right?

I'll rework the PR along these lines and see how it looks, thanks.

Comment (quaid) [reply prow/overlays/cnv-prod/labels_operate-first.yaml:383]: @goern OK, these labels seem _way_ more useful. Did I get the idea correct?

To make it clear for any watchers, what this does is make it so any issue can take an `area` label to mark that it's important to the uses, contributors, or community overall. Similarly, the `kind` labels can mark issues of that kind all over, which can be related or not related to an `area`. This makes those labels more useful to be applied to other areas, e.g. `kind/news` & `area/releases` & `kind/bot` might label an issue that can be brought in as news about updates to a bot.

**Comment by quaid** [comment]
> Well, we mention it https://github.com/thoth-station/core/blob/master/CONTRIBUTING.md#checkerslintersformatters--pre-commit but we have no actual commands provided. It is the same as opf is using...

I got my pre-commit running, and it in fact helped me clean up some mistakes in my commit. I checked out the pull request and updated the patch locally (found some content errors from my last edit of the patch), but I don't have permissions to commit to the PR. :)

I can easily copy over my fixes to the file from the local into the patch via the web UI, but I'll hold off a minute in case there is a better way I can learn about what to do in this case. :) I.e., maybe I should make the change to my fork, and then pull in the changes from the fork to the patch?

**Comment by quaid** [comment]
I have updates in my fork https://github.com/quaid/thoth-application/commit/5a5873b850cecc3cc68a1e5e11228c4187defd08 but then I went sideways and I'm not sure the best fix.

I checked out the main repo locally for the first time, got pre-commit working there, checked out the 'patch-1' pull request, then made my changes from my fork into the 'patch-1', then made additional changes to fix mistakes, then made changes that pre-commit, and finally it's all committed locally. But I can't push to the main repo.

The changes are small and I can just do them somewhere else, like manually via GitHub, but then I can't show I have pre-commit fixed so it passes that test here.

Should I go back locally and make the changes in my quaid/thoth-application fork, then roll those back into the pull request via the web UI?

**Comment by quaid** [comment]
> I have updates in my fork [quaid@5a5873b](https://github.com/quaid/thoth-application/commit/5a5873b850cecc3cc68a1e5e11228c4187defd08) but then I went sideways and I'm not sure the best fix.
> 
> I checked out the main repo locally for the first time, got pre-commit working there, checked out the 'patch-1' pull request, then made my changes from my fork into the 'patch-1', then made additional changes to fix mistakes, then made changes that pre-commit, and finally it's all committed locally. But I can't push to the main repo.
> 
> The changes are small and I can just do them somewhere else, like manually via GitHub, but then I can't show I have pre-commit fixed so it passes that test here.
> 
> Should I go back locally and make the changes in my quaid/thoth-application fork, then roll those back into the pull request via the web UI?

As soon as I wrote above I realized that was the path, and it appears I managed to commit my final version of changes to the PR and get the pre-commit check to pass.

If that's it it, it appears that @tumido needs to do a final review?

File: prow/overlays/cnv-prod/labels_operate-first.yaml
Comment (tumido) [inline prow/overlays/cnv-prod/labels_operate-first.yaml:389]: Should this one replace [`sig/user-experience`](https://github.com/operate-first/support/labels/sig%2Fuser-experience)?

File: prow/overlays/cnv-prod/labels_operate-first.yaml
Comment (tumido) [inline prow/overlays/cnv-prod/labels_operate-first.yaml:135]: Should this one replace [`onboarding`](https://github.com/operate-first/support/labels/onboarding)?

What should we do with [`onboarding-argocd`](https://github.com/operate-first/support/labels/onboarding-argocd)?

**Review by tumido — COMMENTED** [review]
I'm kinda getting a bit lost in all that `sig/NAME` `kind/NAME` `area/NAME` `user/NAME`. Different preposition, same name... I get that it may have a slightly different meaning but unless we all know how to use them properly and actually all of us use them in the same way, it's a mess. @quaid do you have any plan in mind on how we can make it clear for ourselves when to use what?

File: prow/overlays/cnv-prod/labels_operate-first.yaml
Comment (quaid) [reply prow/overlays/cnv-prod/labels_operate-first.yaml:389]: > Should this one replace [`sig/user-experience`](https://github.com/operate-first/support/labels/sig%2Fuser-experience)?

I've been trying to figure out what `sig` means. I see it in the Prow commands but haven't looked much further as it sounded like it might be about special interest groups, which didn't seem to be the fit.

File: prow/overlays/cnv-prod/labels_operate-first.yaml
Comment (quaid) [reply prow/overlays/cnv-prod/labels_operate-first.yaml:135]: That is an interesting nuance. I am thinking about the steps of onboarding individuals into the project, but that includes in some cases onboarding their project code into the infrastructure.

If I get the philosophy correct, should it be:

`kind/onboarding` and `area/cluster`|`area/argocd`

I'm not trying to create a proliferation of labels, my intention here was to try getting most of the labels I think I'll need for labeling issues/PRs across the project for their relation to aspects of the community.

Comment (tumido) [reply prow/overlays/cnv-prod/labels_operate-first.yaml:135]: Yeah, I think the `kind/onboarding` + either `area/cluster` or `area/argocd` would be a good option.

> I'm not trying to create a proliferation of labels, my intention here was to try getting most of the labels I think I'll need for labeling issues/PRs across the project for their relation to aspects of the community.

I'm with you. If we can bring some order to it, it'll be nice :slightly_smiling_face:

**Comment by goern** [comment]
> I'm kinda getting a bit lost in all that `sig/NAME` `kind/NAME` `area/NAME` `user/NAME`. Different preposition, same name... I get that it may have a slightly different meaning but unless we all know how to use them properly and actually all of us use them in the same way, it's a mess. @quaid do you have any plan in mind on how we can make it clear for ourselves when to use what?

Heya, I think `kind` and `area` is pretty straight forward? A bug in deployment_name/zero-prod could be labeled using these. 

`sig/` is a very special thing from the Kubernetes community, see https://github.com/kubernetes/community

`user` ... idk

**Comment by quaid** [comment]
> > I'm kinda getting a bit lost in all that `sig/NAME` `kind/NAME` `area/NAME` `user/NAME`. Different preposition, same name... I get that it may have a slightly different meaning but unless we all know how to use them properly and actually all of us use them in the same way, it's a mess. @quaid do you have any plan in mind on how we can make it clear for ourselves when to use what?
> 
> Heya, I think `kind` and `area` is pretty straight forward? A bug in deployment_name/zero-prod could be labeled using these.
> 
> `sig/` is a very special thing from the Kubernetes community, see https://github.com/kubernetes/community

OK, so it's a way of labeling something as related to a SIG, that makes sense. Are we using `sig/` in the same way, i.e., to refer to work of a project sub-group, team, or sub-project?
 
> `user` ... idk

Yeah, I didn't realize I was creating a whole new namespace within something that has (maintains?) roots with an upstream. So I modified the patch to now have `area/user` and 'area/contributor` (with all the new `area` labels.)

I think that accomplishes my goals without inventing a new namespace.

**Comment by tumido** [comment]
note to ^, that work is tracked at: https://github.com/operate-first/community/issues/36

**Review by tumido — APPROVED** [review]
/approve

**Pull request merged**
